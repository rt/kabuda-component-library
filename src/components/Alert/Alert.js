import withStyles from 'isomorphic-style-loader/lib/withStyles';
import React from 'react';
import PropTypes from 'prop-types';
import s from './Alert.css';
import cx from 'classnames';
import Base from '../Base';

class Alert extends Base {

    constructor(props) {
        super(props);
        
        this.state = {}
        this.handleClose = () => { this.setState( {closed: true} ) };
    }

    render() {
        let cls = null;
        switch (this.props.type) {
            case 'danger':
                cls = cx(s.panel, s.danger);
                break;
        }
        return this.state.closed ? null : (
            <div data-e2e={this.e2e()} className={cls}>
                <span onClick={this.handleClose} className={s.closebtn}>&times;</span>
                {this.props.children}
            </div>
        );
    }
};

Alert.propTypes = {
    type: PropTypes.string,    
    children: PropTypes.node,
};

export default withStyles(s)(Alert);
