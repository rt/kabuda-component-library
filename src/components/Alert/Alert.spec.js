import React from 'react';
import Alert from './Alert';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';

it('renders correctly', () => {
    const tree = shallow(<Alert>PebbleFields</Alert>);

    // const tree = renderer.create(<Alert>PebbleFields</Alert>).toJSON();
    expect(tree).toMatchSnapshot();
});
