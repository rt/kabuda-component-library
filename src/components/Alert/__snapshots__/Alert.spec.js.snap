// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={null}
  data-e2e="alert"
>
  <span
    className="closebtn"
    onClick={[Function]}
  >
    Ã—
  </span>
  PebbleFields
</div>
`;
